File: src/main/java/com/example/backend_qlcv/entity/Lists.java
package com.example.backend_qlcv.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Builder
@Table(name = "lists")
public class Lists {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "position")
    private Integer position;

    @Column(name = "status")
    private Boolean status;

    @ManyToOne
    @JoinColumn(name = "board_id", referencedColumnName = "id", insertable = false, updatable = false)
    private Board board;

    @Column(name = "board_id")
    private Long boardId;
}

File: src/app/core/model/lists/lists.ts
import {Board} from "../board/board";
import {Card} from "../card/card";

export class Lists{
  id: any;
  name: string;
  position: number;
  boardId?: string;
  status: boolean |unknown;
  cards: Card[];
  constructor(id: any, name: any, position: any, cards: Card[]) {
    this.id = id;
    this.name = name;
    this.position = position;
    this.cards = cards || []; // Khởi tạo mảng thẻ, nếu không có sẵn thẻ nào
  }
}
